### PR TITLE
Use port ranges instead of ports when possible.

### PR DESCRIPTION
**Pull request by frankban**
Only single ports outside of ranges are turned into links.
Fixes https://github.com/juju/juju-gui/issues/2596

Also reorganize the view so that public ports are on top.
Fixes https://github.com/juju/juju-gui/issues/2598

Avoid showing "none" when no ports are still available but we've got addresses.
Fixes https://github.com/juju/juju-gui/issues/2307

Additionally, fix the inspector behavior when changing charm versions. Now it displays the new-style charm URLs and clicking on those actually works.
Fixes https://github.com/juju/juju-gui/issues/2570

In order to QA port ranges, you can use the eclipse-che charm, which anyway does not work in LXD. Also, since that charm has many ports open, the WebSocket message is too large and cannot be handled by the WebSocket library used by guiproxy (this will change when guiproxy starts using gorilla websockets). So I guess the best way to QA this is with a gijoe connected to a model in a public cloud.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6918/

File: jujugui/static/gui/src/app/store/env/controller-api.js [hunk 1]
```diff
@@ -1308,6 +1308,7 @@ YUI.add('juju-controller-api', function(Y) {
       Modify (grant or revoke) user access to the specified model.
 
       @method _modifyModelAccess
+      @method _modifyModelAccess
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: duplicate line?

Comment (frankban) [inline jujugui/static/gui/src/app/store/env/controller-api.js]: Fixed, good catch ;-)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6919/

**Comment by hatched** [comment]
Thanks @frankban QA OK on all of the issues.

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**